src/compiler/types.ts

```diff
@@ -1479,7 +1484,7 @@ namespace ts {
         Merged                  = 0x02000000,  // Merged symbol (created during program binding)
         Transient               = 0x04000000,  // Transient symbol (created during type check)
         Prototype               = 0x08000000,  // Prototype property (no source representation)
-        UnionProperty           = 0x10000000,  // Property in union type
+        SyntheticProperty       = 0x10000000,  // Property in union or intersection type

```

Comment:
we use "UnionOrIntersection" elsewhere.  It seems like another place we'd want to use them.  i.e. "UnionOrIntersectionProperty".  "Synthetic" is odd, as it's a term we don't use consistently for this concept elsewhere.